renderer/src/components/ActivityLog.tsx

```diff
@@ -1,37 +1,72 @@
-    const unsubscribe = window.electron.onActivityLogged(allActivities => {
-      setActivities(allActivities)
-    })
-    return unsubscribe
-  }, [])
-
-  // TODO: add proper CSS styling :)
-  const style: React.CSSProperties = {
-    listStyle: 'none',
-    textAlign: 'left',
-    fontSize: '1rem',
-    fontFamily: 'monospace',
-    border: '1px solid white',
-    padding: '1em'
-  }
-
-  return <>
-    <h3>Activity Log</h3>
-    <ul style={style}>
-      {activities.map(activity => <li key={activity.id}>{activity.message}</li>).reverse()}
-    </ul>
-  </>
+import { FC } from 'react'
+import { ActivityEventMessage } from '../typings'
+import dayjs from 'dayjs'
+import { ReactComponent as WarningIcon } from '../assets/img/warning.svg'
+import { ReactComponent as JobIcon } from '../assets/img/job.svg'
+
+const ActivityLogItem: FC<ActivityEventMessage> = (activity) => {
+  const time = new Intl.DateTimeFormat(window.navigator.language, {
+    hour: 'numeric', minute: 'numeric', second: 'numeric'
+  }).format(new Date(activity.timestamp))
+
+  return (
+    <div className="flex py-4 border-b border-grayscale-350 activity-item">
+      <div className="w-10">
+        {activity.type === 'info' &&
+          <i><JobIcon className="btn-icon-primary-small icon-primary"/></i>}
+        {activity.type === 'error' &&
+          <i><WarningIcon className="btn-icon-primary-small icon-error"/></i>}
+      </div>
+      <div className="flex-1 mr-4 text-body-s">
+        <p>{activity.message}</p>
+      </div>
+      <div className="">
+        <span className='number text-body-2xs text-grayscale-600'>{time}</span>
+      </div>
+    </div>
+  )
+}
+
+interface DateSeparatorProps {
+  date: string,
+}
+
+const DateSeparator: FC<DateSeparatorProps> = ({ date }) => {
+  const today = dayjs().format('YYYY-MM-DD')
+  const yesterday = dayjs().subtract(1, 'day').format('YYYY-MM-DD')
+  const displayStr = date === today ? 'today' : date === yesterday ? 'yesterday' : dayjs(date).format('D MMM')
+
+  return <div className="uppercase text-body-3xs mb-4">{displayStr}</div>
 }
+
+interface ActivitiesByDate {
+  [date: string]: ActivityEventMessage[];
+}
+interface ActivityLogProps {
+  activities: ActivityEventMessage[];
+}
+
+const ActivityLog: FC<ActivityLogProps> = ({ activities = [] }) => {
+  const activitiesByDate: ActivitiesByDate =
+    activities
+      .sort((x, y) => y.timestamp - x.timestamp)
+      .reduce((groups: ActivitiesByDate, activity: ActivityEventMessage) => {
+        const date = dayjs(activity.timestamp).format('YYYY-MM-DD')
+        return { ...groups, [date]: groups[date] ? groups[date].concat(activity) : [activity] }
+      }, {})
+
+  return (
+    <div className='activity-log'>
+      {Object.keys(activitiesByDate).map((date) => (
+        <div key={date} className="mb-12">
+          <DateSeparator date={date}/>
+          <div>
+            {activitiesByDate[date].map((activity) => <ActivityLogItem key={activity.id} {...activity} />)}
```

Comment:
the spread operator can lead to unepexted results with React component, can we please pass all properties explicitly?

Reply:
[6800270](https://github.com/filecoin-project/filecoin-station/pull/144/commits/68002702b9462d2925937c413ca7e406441a231b)